frontend/src/lib/utils.ts

```diff
@@ -203,3 +203,20 @@ export function labelVisibilityProtoValueToEnum(
       return LabelVisibilityOptions.Visible
   }
 }
+
+export function getHTMLIFrameElement(): HTMLIFrameElement | null {
```

Comment:
A few things:
* I think we should call `isInChildFrame()` here and return `null` if it returns false so that the code in `App.tsx` and elsewhere doesn't have to worry about doing this
* A name like `getEnclosingIFrame`/`getIFrameEnclosingApp` (or something like that) would be a bit more descriptive 
* There's an edge case here where a web page that embeds a Streamlit app has other iframes too. In this case, it would be possible to open the print dialog for the wrong iframe depending on how they're ordered (this should be rare, but I think it happens on our docs site). We'd need to figure out a way to double-check that the iframe we've grabbed is the right one (maybe by verifying that `iframe.contentWindow.window == window`?)

I also think it might be possible to write unit tests for this function by mounting a few `<App/>` components within an `iframe`, but I'd be okay with just manually testing this by embedding a few streamlit apps in a small wrapper app in case testing turns out to be unreasonably difficult.

Reply:
> I think we should call isInChildFrame() here and return null if it returns false so that the code in App.tsx and elsewhere doesn't have to worry about doing this

It's done 👍 

> A name like getEnclosingIFrame/getIFrameEnclosingApp (or something like that) would be a bit more descriptive

It's renamed to `getIFrameEnclosingApp` 👍 

> There's an edge case here where a web page that embeds a Streamlit app has other iframes too. In this case, it would be possible to open the print dialog for the wrong iframe depending on how they're ordered (this should be rare, but I think it happens on our docs site). We'd need to figure out a way to double-check that the iframe we've grabbed is the right one (maybe by verifying that iframe.contentWindow.window == window?) / will probably have to iterate through the document's iframe tags to search for the correct one

True, partially it's done with the selector `document.querySelectorAll('iframe[title="streamlitApp"]')` which checks for `title="streamlitApp"`, however we can have multiple iFrames with this title, on Streamlit Cloud every iFrame has this tag, however I think that IFrame creation is done on the provider side, so somehow selector needed to be passed into the Streamlit app, I am not sure how to achieve that 🤔 

> I also think it might be possible to write unit tests for this function by mounting a few <App/> components within an iframe, but I'd be okay with just manually testing this by embedding a few streamlit apps in a small wrapper app in case testing turns out to be unreasonably difficult.

Just tried to do it with Jest/Enzyme and it's really tricky to do, just added one test in Unit tests which check if `printCallback` method is defined on `App` component. I guess the best way to test it is to write E2E tests, but I'd prefer to stick to manual testing if you do not mind.

@vdonato thank you for taking care of this, what do you think ❓ should we go with the current state or look for some workaround here ❓ also @jrieke and @sfc-gh-mnowotka can you please provide us some input here?

Reply:
Can we see how many iframes there are on the page? If we cannot check which iframe belongs to the app itself, then a workaround could be that we only trigger this printing behavior if there's only one iframe on the page. If there are multiple iframes on the page, we don't trigger our special printing behavior but just open the print dialog on the outer page as normal... Not the cleanest solution but should work for 99% of cases.

Reply:
Sure 👍 will do 👍